### PR TITLE
Phase2-hgx346B Next step toward calibration cell collection for HGCal in the next version of HGCal geometry replacing #43045

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Next step toward calibration cell collection for HGCal in the next version of HGCal geometry replacing #43045

#### PR validation:

Use the runTheMatrix test workflows 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch @srimanob Please approve this PR which is a replacement of an earlier PR

File: SimG4CMS/Calo/src/CaloSD.cc [hunk 1]
```diff
@@ -28,7 +28,7 @@
 #include <memory>
 #include <sstream>
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (civanch) [inline SimG4CMS/Calo/src/CaloSD.cc:31]: @bsunanda , please, comment out the DEBUG

File: SimG4CMS/Calo/src/CaloSD.cc [hunk 2]
```diff
@@ -169,6 +170,23 @@ CaloSD::CaloSD(const std::string& name,
 
 CaloSD::~CaloSD() {}
```
Comment (civanch) [inline SimG4CMS/Calo/src/CaloSD.cc:171]: should it be "default"?

Comment (bsunanda) [reply SimG4CMS/Calo/src/CaloSD.cc:171]: This change causes some compilation issues. I leave this for the time being.

File: SimG4CMS/Calo/src/CaloSD.cc (hunk 1 above)
Comment (bsunanda) [reply SimG4CMS/Calo/src/CaloSD.cc]: Done

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@srimanob Please approve this

**Comment by srimanob** [comment]
+Upgrade

**Comment by rappoccio** [comment]
+1

**Pull request merged**